### PR TITLE
Agent Memory SDKs: allow for custom schemas

### PR DESCRIPTION
**Pull request by bbqiu**
## Problem

When using the `public` schema, SPs need to be granted postgres permissions explicitly. this means that when creating a template through the UI (without any manual permission grants in the UI / through a script), the SP will not have permission to write/create on it.

## Solution
Adds a `schema` param to CheckpointSaver, AsyncCheckpointSaver, DatabricksStore, AsyncDatabricksStore, and AsyncDatabricksSession. Since it'll be a new creation from the UI (given a fresh db), this will hopefully bypass permission issues.

The implementation uses two layers:
- Pool/engine level: SET search_path on each connection
- Class level: CREATE SCHEMA IF NOT EXISTS in setup()/ensure_tables()

## Rollout 
For current users of this SDK, nothing changes since this param defaults to None.

For new users of the SDK (like the updated templates) that provide a schema, this will create tables in the specified schema instead of `public`

## Testing
unit tests
manually tested with templates

### CODE REVIEW

File: integrations/openai/src/databricks_openai/agents/session.py [hunk 1]
```diff
@@ -168,25 +173,40 @@ def __init__(
...
 
+    async def _ensure_tables(self) -> None:
+        """Ensure schema and tables are created before any database operations."""
+        if self._schema and self._create_tables:
+            import re
+
+            from sqlalchemy import text
+
+            if not re.match(r"^[A-Za-z_][A-Za-z0-9_]*$", self._schema):
+                raise ValueError(f"Invalid schema name: {self._schema!r}")
```
Comment (dhruv0811) [inline integrations/openai/src/databricks_openai/agents/session.py:184]: very nit: can we add a better desc of what the name should look like? if the error is thrown

File: src/databricks_ai_bridge/lakebase.py [hunk 2]
```diff
@@ -441,6 +445,20 @@ def connect(cls, conninfo: str = "", **kwargs):
...
         timeout = pool_kwargs.pop("timeout", DEFAULT_TIMEOUT)
 
+        # Set search_path on each connection when a schema is specified
+        user_configure = pool_kwargs.pop("configure", None)
+        _configure_fn = None
+        if self.schema:
+            _schema = self.schema
+            _user_configure = user_configure
+
+            def _configure_fn(conn):  # type: ignore[misc]  # noqa: F811
```
Comment (dhruv0811) [inline src/databricks_ai_bridge/lakebase.py:455]: Same function duplicated like thrice across the file it seems, can we extract this to have only one copy.

File: src/databricks_ai_bridge/lakebase.py [hunk 3]
```diff
@@ -441,6 +445,20 @@ def connect(cls, conninfo: str = "", **kwargs):
         max_size = pool_kwargs.pop("max_size", DEFAULT_MAX_SIZE)
         timeout = pool_kwargs.pop("timeout", DEFAULT_TIMEOUT)
 
+        # Set search_path on each connection when a schema is specified
+        user_configure = pool_kwargs.pop("configure", None)
```
Comment (dhruv0811) [inline src/databricks_ai_bridge/lakebase.py:449]: This pop is run unconditionally. User provided configure is silently dropped when no schema is set. I think this violates our "nothing changes for current users" rollout claim. Maybe also add a test covering `configure=fn, schema=None`.

File: integrations/openai/src/databricks_openai/agents/session.py [hunk 4]
```diff
@@ -168,25 +173,40 @@ def __init__(
...
         )
 
+    async def _ensure_tables(self) -> None:
+        """Ensure schema and tables are created before any database operations."""
+        if self._schema and self._create_tables:
+            import re
+
+            from sqlalchemy import text
+
+            if not re.match(r"^[A-Za-z_][A-Za-z0-9_]*$", self._schema):
```
Comment (dhruv0811) [inline integrations/openai/src/databricks_openai/agents/session.py:183]: Why aren't we using `sql.Identifier(...)` like elsewhere in this PR?

File: integrations/openai/src/databricks_openai/agents/session.py [hunk 5]
```diff
@@ -168,25 +173,40 @@ def __init__(
             session_id,
         )
 
+    async def _ensure_tables(self) -> None:
+        """Ensure schema and tables are created before any database operations."""
+        if self._schema and self._create_tables:
+            import re
+
+            from sqlalchemy import text
```
Comment (dhruv0811) [inline integrations/openai/src/databricks_openai/agents/session.py:181]: Can we move imports to be at the top?

**Review by dhruv0811 — COMMENTED** [review]
Looks good overall! Left some comments 

Also, even with these changes, when I tried it out, I hit a SP-permissions wall in the Drizzle (TypeScript frontend) migration. So just these changes wont be sufficient to get rid of the grant script for our advanced templates right?

File: integrations/langchain/src/databricks_langchain/checkpoint.py [hunk 6]
```diff
@@ -95,10 +110,23 @@ def __init__(
...
+        """Set up the checkpoint database asynchronously, creating the schema if specified."""
+        if self._schema:
+            from psycopg import sql
+
+            async with self._lakebase.connection() as conn:
+                await conn.execute(
+                    sql.SQL("CREATE SCHEMA IF NOT EXISTS {}").format(sql.Identifier(self._schema))
+                )
+        await super().setup()
+
```
Comment (jennsun) [inline integrations/langchain/src/databricks_langchain/checkpoint.py:117]: Rather than grabbing connections and executing SQL queries manually like this, can we update the LakebaseClient in this PR to support `create_schema` (similar to `create_role`) and use it whenever we call this updated setup() function)? 

https://github.com/databricks/databricks-ai-bridge/blob/main/src/databricks_ai_bridge/lakebase.py#L788

Comment (bbqiu) [reply integrations/langchain/src/databricks_langchain/checkpoint.py:117]: great suggestion, added

File: src/databricks_ai_bridge/lakebase.py (hunk 2 above)
Comment (bbqiu) [reply src/databricks_ai_bridge/lakebase.py:461]: abstracted a bit more of it away, but unfortunately need to keep separate copies bc of sync / async

File: integrations/openai/src/databricks_openai/agents/session.py (hunk 1 above)
Comment (bbqiu) [reply integrations/openai/src/databricks_openai/agents/session.py]: removed this regex and used sql identifier instead

**Review by dhruv0811 — APPROVED** [review]
lgtm!
